File: python/1110.py
while True:
    n = int(input())
    
    if n == 0:
        break
    
    arr = [i for i in range(n, 0, -1)]
    discardedArr = []
    
    while len(arr) >= 2:
        x = len(arr) - 1
        discardedArr.append(arr.pop(x))
        y = arr.pop(x - 1)
        arr.insert(0, y)
    
    discardedArrStr = ''    
    
    for x in discardedArr:
        discardedArrStr += str(x) + ', ' 
        
    print(f'Discarded cards: {discardedArrStr[:len(discardedArrStr) - 2]}')
    print(f'Remaining card: {arr[0]}')

File: lua/1110.lua
while true do
    local n = io.read('n')
    if n == 0 then
        break
    end
    local stack = {}
    for i=1, n do
        table.insert(stack, i)
    end
    local discardedCards = {}
    while #stack > 1 do
        local discardedCard = table.remove(stack, 1)
        table.insert(discardedCards, discardedCard)
        local temp = table.remove(stack, 1)
        table.insert(stack, #stack + 1, temp)
    end
    local text = 'Discarded cards:'
    for ind, card in ipairs(discardedCards) do
        if ind > 1 then
            text = text..string.format(', %i', card)
        else
            text = text..string.format(' %i', card)
        end
    end
    print(text)
    print('Remaining card: '..tostring(stack[1]))
end